### ISSUE TITLE
[UX] Make sure "Plain text" does not appear in the text format drop-down

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#1186]
The "Plain text" option for text formats was never meant to be displayed. It's meant to be the default - the one that shows up when you don't have access to anything better - and should not be selectable.

Since most end-users don't fully understand text formats (can't blame them!) we should make this interface less confusing, and fewer options here will help that.

**Comment by quicksketch** [comment]
The PR at https://github.com/backdrop/backdrop/pull/1059 looks really good. It looks as though it would work in all situations. There's an issue that the fallback format is configurable however (though not through the UI) see https://github.com/backdrop/backdrop/pull/1059/files#r39467899.

Tests also need to be updated of course.

I'm not entirely certain about squeezing this into 1.2.0, as much as I like the idea.

**Comment by quicksketch** [comment]
We may also want to incorporate https://github.com/backdrop/backdrop-issues/issues/1092 into this issue.

**Comment by jenlampton** [comment]
Getting tests passing, and incorporated changes from https://github.com/backdrop/backdrop-issues/issues/1092.

**Comment by quicksketch** [comment]
The current PR doesn't work quite like I would expect. There's still a select list even when only one format is available:

![selection_002](https://cloud.githubusercontent.com/assets/100206/9888769/20e63b1c-5bb4-11e5-8a33-c4c6287e8a5f.png)

I would expect there would be no select list if only one format is available.

**Comment by quicksketch** [comment]
Hm, and if you hide the select list entirely, CKEditor disappears because it doesn't know what format is on the page any more. I'll give this some work and see if we can get it going.

**Comment by quicksketch** [comment]
Okay I merged in https://github.com/backdrop/backdrop/issues/1064. This went a little further than I had originally intended:

- I couldn't bear the long filter tips link any longer. Consulting with @jenlampton we concurred that they just didn't belong any more in the age of a rich-text editor. We have more work to do there still in #1089.
- After removing the long filter tips link, we had a lot of room to simplify our filter CSS.
- The styling for Bartik filter tips was wonky and had a lot of unnecessary margins. I removed the excess CSS here as well.
- The drop-down for text format now no longer shows up if you only have access to a single format.

Screenshots:

Drop down:
![selection_003](https://cloud.githubusercontent.com/assets/100206/9892992/60514de2-5bd1-11e5-8053-0f5e8c3c4083.png)

No filter tips long link:
![screenshot21](https://cloud.githubusercontent.com/assets/100206/9892997/6cd13d20-5bd1-11e5-8110-3d95a48443cd.png)

As an authenticated user with only one format:
![screenshot22](https://cloud.githubusercontent.com/assets/100206/9893007/7f1e3d0c-5bd1-11e5-911a-acfc3d617b80.png)

I'd love to see the rather unnecessary auto-link help go away somewhere else, but we can save that for #1089.
